src/aro/SymbolStack.zig

```diff
@@ -31,72 +33,72 @@ pub const Kind = enum {
     constexpr,
 };
 
-const SymbolStack = @This();
+scopes: std.ArrayListUnmanaged(Scope) = .{},
+innermost: usize = 0,
 
-syms: std.MultiArrayList(Symbol) = .{},
-scopes: std.ArrayListUnmanaged(u32) = .{},
+const Scope = struct {
+    vars: std.AutoHashMapUnmanaged(StringId, Symbol) = .{},
+    tags: std.AutoHashMapUnmanaged(StringId, Symbol) = .{},
+
+    fn deinit(self: *Scope, allocator: Allocator) void {
+        self.vars.deinit(allocator);
+        self.tags.deinit(allocator);
+    }
+
+    fn clearRetainingCapacity(self: *Scope) void {
+        self.vars.clearRetainingCapacity();
+        self.tags.clearRetainingCapacity();
+    }
+};
 
 pub fn deinit(s: *SymbolStack, gpa: Allocator) void {
-    s.syms.deinit(gpa);
+    std.debug.assert(s.innermost == std.math.maxInt(@TypeOf(s.innermost)));
```

Comment:
Could you instead change `get` to use `innermost - 1` and remove the wrapping from `pop`?

Reply:
I've been coding on a plane with 2s latency so it's possible I messed up the benchmark; I'll repeat it when I get home.

Updated SymbolStack to remove the wrapping and renamed the field to `active_len` since I think that more accurately describes its role after the change.